pkg/generate/code/set_sdk.go

```diff
@@ -243,14 +226,37 @@ func SetSDK(
 		if found {
 			sourceAdaptedVarName += cfg.PrefixConfig.SpecField
 		} else {
-			f, found = r.StatusFields[memberName]
+			f, found = r.StatusFields[renamedName]
 			if !found {
 				// TODO(jaypipes): check generator config for exceptions?
 				continue
 			}
 			sourceAdaptedVarName += cfg.PrefixConfig.StatusField
 		}
 		sourceAdaptedVarName += "." + f.Names.Camel
+		sourceFieldPath := f.Names.Camel
+
+		if r.IsSecretField(memberName) {
+			out += setSDKForSecret(
+				cfg, r,
+				memberName,
+				targetVarName,
+				sourceAdaptedVarName,
+				indentLevel,
+			)
+			continue
+		}
+
+		if r.IsSecretField(memberName) {
+			out += setSDKForSecret(
+				cfg, r,
+				memberName,
+				targetVarName,
+				sourceAdaptedVarName,
+				indentLevel,
+			)
+			continue
+		}
```

Comment:
Duplicate code :)

Reply:
oh, wow... copy pasta. Will fix ASAP. thank you!